scripts/variantstore/wdl/extract/gvs_avros_to_vds.py

```diff
@@ -313,35 +324,53 @@ def add_reference_allele(mt):
+                                     truth_sensitivity_snp_threshold=truth_sensitivity_snp_threshold,
+                                     truth_sensitivity_indel_threshold=truth_sensitivity_indel_threshold)
+
+
+            sorted_tranche_data = hl.sorted(vd.tranche_data, key=lambda x: x.truth_sensitivity)
+            vd = vd.annotate_globals(snp_vqslod_threshold=
+                                     sorted_tranche_data.filter(lambda x: (x.model == 'SNP') & (
+                                                 x.truth_sensitivity >= truth_sensitivity_snp_threshold))
+                                     .head().min_vqslod
+                                     ,
+                                     indel_vqslod_threshold=sorted_tranche_data.filter(lambda x: (x.model == 'INDEL') & (
+                                                 x.truth_sensitivity >= truth_sensitivity_indel_threshold))
+                                     .head().min_vqslod
+                                     )
+
+            is_snp = vd.alleles[1:].map(lambda alt: hl.is_snp(vd.alleles[0], alt))
+            vd = vd.annotate_rows(
+                allele_NO=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'N', False)),
+                allele_YES=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'Y', True)),
+                allele_is_snp=is_snp,
+                allele_OK=hl._zip_func(is_snp, vd.alleles[1:],
+                                       f=lambda is_snp, alt:
+                                       hl.coalesce(vd.as_vqsr.get(alt).vqslod >=
+                                                   hl.if_else(is_snp, vd.snp_vqslod_threshold, vd.indel_vqslod_threshold),
+                                                   True))
+            )
+        else:
+            vd = vd.annotate_globals(truth_sensitivity_snp_threshold=truth_sensitivity_snp_threshold,
+                                     truth_sensitivity_indel_threshold=truth_sensitivity_indel_threshold)
+
+            is_snp = vd.alleles[1:].map(lambda alt: hl.is_snp(vd.alleles[0], alt))
+            vd = vd.annotate_rows(
+                allele_NO=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'N', False)),
+                allele_YES=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'Y', True)),
+                allele_is_snp=is_snp,
+                allele_OK=hl._zip_func(is_snp, vd.alleles[1:],
+                                       f=lambda is_snp, alt:
+                                       hl.coalesce(vd.as_vqsr.get(alt).calibration_sensitivity <=
+                                                   hl.if_else(is_snp, vd.truth_sensitivity_snp_threshold, vd.truth_sensitivity_indel_threshold),
+                                                   True))
```

Comment:
likewise I think this is the same as the block from 344 to 355 delta the `coalesce` predicate

Reply:
Actually it's operating on the 'calibration_sensitivity' here (and the directionality of the diff is different).